### PR TITLE
Do Lesson - Result Page

### PR DESCRIPTION
**Pull request by truongnv0211**

### CODE REVIEW

File: app/controllers/lessons_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  def edit
+    @lesson = Lesson.find params[:id]
+  end
+
+  def create
+    @lesson = current_user.lessons.build category_id: params[:category_id]
+    if @lesson.save
+      redirect_to edit_lesson_path(@lesson.id)
```
Comment (ducminhvu) [inline app/controllers/lessons_controller.rb]: ()

File: app/controllers/lessons_controller.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    end
+  end
+
+  def update
+    @lesson = Lesson.find params[:id]
+    if @lesson.update_attributes params_lesson
+      redirect_to lesson_path @lesson
+    else
+      flash[:error] = t :lesson_update_error
+      redirect_to categories_path()
```
Comment (ducminhvu) [inline app/controllers/lessons_controller.rb]: ()

File: app/models/lesson.rb [hunk 3]
```diff
@@ -3,7 +3,18 @@ class Lesson < ActiveRecord::Base
...
+  accepts_nested_attributes_for :lesson_words
+  after_create :random_words
 
   scope :lessons_learned, ->user{where Settings.sql_query.lessons_learned, user.id}
   scope :following_leaned, ->user{where Settings.sql_query.following_leaned, user.id, user.id}
+
+  private
+  def random_words
+    category = Category.find self.category_id
+    category.words.sample(20).each do |word|
```
Comment (ducminhvu) [inline app/models/lesson.rb]: settings

File: app/models/category.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 class Category < ActiveRecord::Base
   has_many :words, dependent: :destroy
   has_many :lessons, dependent: :destroy
+  accepts_nested_attributes_for :words
```
Comment (dieunb) [inline app/models/category.rb]: no need

File: app/models/lesson.rb [hunk 5]
```diff
@@ -3,7 +3,18 @@ class Lesson < ActiveRecord::Base
...
   belongs_to :category
+  accepts_nested_attributes_for :lesson_words
+  after_create :random_words
 
   scope :lessons_learned, ->user{where Settings.sql_query.lessons_learned, user.id}
   scope :following_leaned, ->user{where Settings.sql_query.following_leaned, user.id, user.id}
+
+  private
+  def random_words
+    category = Category.find self.category_id
```
Comment (dieunb) [inline app/models/lesson.rb]: try `category = self.category`

File: app/models/lesson.rb (hunk 3 above)
Comment (dieunb) [inline app/models/lesson.rb]: refactor this block code

File: app/views/lessons/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<div class="row">
+    <h3><%= @lesson.category.title %></h3>
```
Comment (dieunb) [inline app/views/lessons/edit.html.erb]: indent

File: app/models/lesson_word.rb [hunk 7]
```diff
@@ -1,4 +1,7 @@
 class LessonWord < ActiveRecord::Base
   belongs_to :lesson
   belongs_to :word
+  belongs_to :answer
+
+  scope :answered, ->{where Settings.sql_query.answered , :answer}
```
Comment (khanhkhanh) [inline app/models/lesson_word.rb]: Remove redundant whitespace(s)

File: app/views/lesson_words/_lesson_word.html.erb [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<tr>
+  <td><span class="<%= lesson_word.answer.correct ?
```
Comment (khanhkhanh) [inline app/views/lesson_words/_lesson_word.html.erb]: lesson_word.answer.correct => lesson_word.answer.correct?

File: app/views/lessons/_word.html.erb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<div class="answers">
+  <h3 class="col-md-6"><%= f.object.word.jp_word %></h3>
+  <div class="col-md-6">
+    <%= f.collection_radio_buttons :answer_id, f.object.word.answers,
+      :id, :answer_content do |item| %>
+      <%= item.label(class: "answer-selected btn btn-block btn-default") {
+        item.radio_button(class: "hidden") + item.text } %>
```
Comment (khanhkhanh) [inline app/views/lessons/_word.html.erb]: Remove redundant whitespace(s)

File: app/views/lessons/_word.html.erb [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+    <% end %>
+</div>
```
Comment (khanhkhanh) [inline app/views/lessons/_word.html.erb]: Indentation

File: app/views/lessons/edit.html.erb [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    <%= form_for @lesson, html: {class: "col-md-12"}  do |f| %>
+      <%= f.fields_for :lesson_words  do |builder| %>
+        <%= render "word", f: builder %>
+      <% end %>
+      <div class="col-md-2">
+      <%= f.submit t("result"), class: "btn btn-primary submit-answer hidden" %>
+      </div>
+    <% end %>
+</div>
+<script type="text/javascript">
```
Comment (khanhkhanh) [inline app/views/lessons/edit.html.erb]: Move these script lines into application.js

File: app/views/lessons/show.html.erb [hunk 12]
```diff
@@ -0,0 +1,13 @@
+<div class="container">
+  <h4><%= @lesson.category.title %></h4>
+  <table class="table table-striple">
+  <tr>
+    <th></th>
+    <th><%= t(:jp_word) %></th>
```
Comment (khanhkhanh) [inline app/views/lessons/show.html.erb]: Remove ()

File: app/views/lessons/show.html.erb [hunk 13]
```diff
@@ -0,0 +1,13 @@
+<div class="container">
+  <h4><%= @lesson.category.title %></h4>
+  <table class="table table-striple">
+  <tr>
```
Comment (khanhkhanh) [inline app/views/lessons/show.html.erb]: Indentation

File: app/views/categories/_category.html.erb [hunk 14]
```diff
@@ -1,7 +1,8 @@
 <div>
   <h3><%= category.title %></h3>
-  <i><%= Word.located_in(category).count %> - <%= t :words %></i>
+  <i><%= category.words.count %> - <%= t :words %></i>
   <p><%= category.description %></p>
-  <%= link_to t("start"), "#", class: "btn btn-sm pull-right btn-primary" %><br />
+  <%= link_to t("start"), lessons_path(category_id: category.id),
+   method: :POST, class: "btn btn-sm pull-right btn-primary" %><br />
```
Comment (khanhkhanh) [inline app/views/categories/_category.html.erb]: Need one more whitespace before method

File: app/views/lessons/edit.html.erb [hunk 15]
```diff
@@ -0,0 +1,16 @@
+<div class="row">
+    <h3><%= @lesson.category.title %></h3>
+    <h4><span class="answer_counting"><%= t :answered %></span></h4>
+    <%= form_for @lesson, html: {class: "col-md-12"}  do |f| %>
```
Comment (khanhkhanh) [inline app/views/lessons/edit.html.erb]: Remove redundant whitespace(s)

File: app/views/lessons/edit.html.erb [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+      <%= f.fields_for :lesson_words  do |builder| %>
```
Comment (khanhkhanh) [inline app/views/lessons/edit.html.erb]: Remove redundant whitespace(s)

File: app/views/lessons/edit.html.erb (hunk 11 above)
Comment (truongnv0211) [inline app/views/lessons/edit.html.erb]: Not allway put js in file. http://stackoverflow.com/questions/14328449/when-do-you-put-javascript-in-body-when-in-head-and-when-use-doc-load

**Pull request merged**